src/Eval.hs

```diff
@@ -1048,10 +1048,14 @@ specialCommandSet ctx [orig@(XObj (Sym path@(SymPath _ _) _) _ _), val] =
     evalAndSet :: Binder -> (Context -> Env -> Either EvalError XObj -> Binder -> IO (Context, Either EvalError XObj)) -> Env -> IO (Context, Either EvalError XObj)
     evalAndSet binder setter env =
       case xobjTy (binderXObj binder) of
-        -- don't type check dynamic bindings
+        -- don't type check dynamic or untyped bindings
+        -- TODO: Figure out why untyped cases are sometimes coming into set!
         Just DynamicTy ->
           evalDynamic ResolveLocal ctx val
             >>= \(newCtx, result) -> setter newCtx env result binder
```

Comment:
Since the two cases for `Just DynamicTy` and `Nothing` are the same, should we pull them into a `where`?

Reply:
Nice catch, done!